### ISSUE TITLE
Display Icons for Custom Code Host URLs

### ISSUE THREAD

**Issue by gabtorre** [sourcegraph/sourcegraph#60763]
#### Feature request description

<img width="64" alt="Screenshot 2024-02-27 at 2 12 22 PM" src="https://github.com/sourcegraph/sourcegraph/assets/69164745/43f2b180-9336-4f30-acf3-88d6343fe923">

Currently, search results only display icons for cloud code hosts, leading to confusion when navigating repositories with identical names across different code hosts. This feature request proposes the extension of icon display functionality to include custom or corporate code host URLs, such as corp.github.com or ghe.corp.com.

#### Is your feature request related to a problem? If so, please describe.

Users have reported difficulty distinguishing between repositories with the same name but hosted on different code hosts.

**Comment by camdencheek** [comment]
Yeah, I noticed this as well when doing some related work. It's pretty jarring, especially for instances that have a lot of non-github.com code

**Comment by camdencheek** [comment]
At the very least, we should have a placeholder icon that the user can hover over to get the full host name
